wordpress-mobile/WordPress-iOS
# Issue/9476 truncated buttons

Fixes #9476 

![non-truncated-buttons](https://user-images.githubusercontent.com/9772967/40948309-5b27e284-682d-11e8-90c7-cf88ad9156d8.png)

The `Help` button was inside a view with a hardcoded size of 44x44
<img width="285" alt="screen shot 2018-06-04 at 10 09 25 am" src="https://user-images.githubusercontent.com/9772967/40948337-8d59da0a-682d-11e8-9878-0718fd944427.png">

So, any text inside that box that was long enough would be truncated.
This was fixed by letting the width grow with the button, but it will never be smaller than 44 points wide.
<img width="282" alt="screen shot 2018-06-04 at 10 07 56 am" src="https://user-images.githubusercontent.com/9772967/40948387-cfe1f920-682d-11e8-8254-ac970dd44f46.png">

Because of these changes, the position of the Zendesk notification indicator had to be modified a bit.
I tried to keep it as similar as it was before:

![simulator screen shot - iphone 5 - 2018-06-04 at 19 44 20](https://user-images.githubusercontent.com/9772967/40948752-d4ce518e-682f-11e8-9fb8-70bd5b9a3afd.png)

I took the opportunity to refactor a bit the `addHelpButtonToNavController` method, hopefully the name changes are fine.

@ScoutHarris could you please give me a hand with this? Thank you!

To test:
- Change the language of your iPhone 5 device or simulator (iOS 10) to Spanish.
- Run the app on the device or simulator.
- Log out if you were logged in.
- Go to `Log In`
- Log in by entering your site address.
- Enter a site address and `Next`
- Check that the `Help` (Ayuda) and the `Next` (Siguiente) buttons display correctly.

**NOTE**: For some reason, in the simulator the translation wasn't complete for me. If this happens to you, you can manually change the strings in code to reproduce the issue:

- `LoginSelfHostedViewController`: `Lost your password?` to `Has perdido tu contraseña?"
- `NUXViewControllerBase`: `Help` -> `Ayuda`
(Any long string will do)
